lib/cocoapods/installer.rb

```diff
@@ -163,15 +160,25 @@ def download_dependencies
       end
     end
 
-    def generate_pods_project
+    #-------------------------------------------------------------------------#
+
+    # @!group Pods Project Generation
+
+    private
+
+    def create_generator
+      Xcode::PodsProjectGenerator.new(aggregate_targets, sandbox, pod_targets, analysis_result, installation_options, config)
+    end
+
+    # Generate the 'Pods/Pods.xcodeproj' project.
+    #
+    def generate_pods_project(generator = create_generator)
       UI.section 'Generating Pods project' do
-        prepare_pods_project
```

Comment:
The changes here maintain the previous method call ordering, but inside `PodsProjectGenerator`